packages/cozy-harvest-lib/src/services/updateAccountsPassword.js

```diff
@@ -0,0 +1,61 @@
+import SymmetricCryptoKey from 'cozy-keys-lib/transpiled/SymmetricCryptoKey'
+import get from 'lodash/get'
+import set from 'lodash/set'
+import unset from 'lodash/unset'
+
+const updateAccountsPassword = async (
+  cozyClient,
+  vaultClient,
+  bitwardenCipherDocument
+) => {
+  const cozyKeys = await cozyClient
+    .getStackClient()
+    .fetchJSON('GET', '/bitwarden/organizations/cozy')
+
+  const encryptedPassword = get(bitwardenCipherDocument, 'login.password')
+  const bitwardenCipherId = get(bitwardenCipherDocument, '_id')
+
+  const [encTypeAndIv, data, mac] = encryptedPassword.split('|')
+  const [encTypeString, iv] = encTypeAndIv.split('.')
+  const encType = parseInt(encTypeString)
+
+  const orgKey = new SymmetricCryptoKey(
+    vaultClient.Utils.fromB64ToArray(cozyKeys.organizationKey),
+    encType
+  )
+
+  const decryptedPassword = await vaultClient.cryptoService.aesDecryptToUtf8(
+    encType,
+    data,
+    iv,
+    mac,
+    orgKey
+  )
+
+  const accounts = await cozyClient.query(
+    cozyClient
+      .find('io.cozy.accounts')
+      .where({
+        'relationships.vaultCipher': {
+          _id: bitwardenCipherId,
+          _type: 'com.bitwarden.ciphers'
+        }
+      })
+      .indexFields(['relationships.vaultCipher._id'])
+  )
+
+  await Promise.all(
+    accounts.data.map(account => {
+      set(account, 'auth.password', decryptedPassword)
```

Comment:
I think you also need to decrypt the `username` from the cipher and put it in the `auth.login` field. As I understand, login and password are encrypted together in a single blob (the `credentials_encrypted`), and if the password is sent without the login, the login will be lost after the update.

Reply:
Well, I have read again that part of the code in the stack, and it looks like the login+password are encrypted in the `credentials_encrypted` field, but the login is also kept in clear (I don't understand why it has been made this way). So, it should work even you don't set `auth.login` in your code.

Reply:
Yeah it's also what I observed, but @edas made the same comment below, so I'll update the login with the cipher just in case.